File: MB-Secretaria/confidence/confidence-score.ts
import type { EvidenceEvaluation, HallucinationRisk, ProviderConsistencyResult } from "./types";

export function calculateConfidenceScore(input: {
  evidence: EvidenceEvaluation;
  providerConsistency: ProviderConsistencyResult;
  uncertaintyScore: number;
  hallucinationRisk: HallucinationRisk;
}): number {
  let score = input.evidence.evidence_completeness * 0.4 + input.providerConsistency.consistency_score * 0.35 + (1 - input.uncertaintyScore) * 0.25;
  if (input.hallucinationRisk.risk_level === "high") score -= 0.15;
  if (input.hallucinationRisk.risk_level === "critical") score -= 0.25;
  return Number(Math.max(0, Math.min(1, score)).toFixed(3));
}

File: MetaBrain/confidence_py/confidence_score.py
from .types import EvidenceEvaluation, HallucinationRisk, ProviderConsistencyResult


def calculate_confidence_score(
    evidence: EvidenceEvaluation,
    provider_consistency: ProviderConsistencyResult,
    uncertainty_score: float,
    hallucination_risk: HallucinationRisk,
) -> float:
    score = evidence.evidence_completeness * 0.4 + provider_consistency.consistency_score * 0.35 + (1 - uncertainty_score) * 0.25
    if hallucination_risk.risk_level == "high":
        score -= 0.15
    if hallucination_risk.risk_level == "critical":
        score -= 0.25
    return round(max(0.0, min(1.0, score)), 3)
